bee-api/bee-rest-api/src/endpoints/routes/api/v2/info.rs

```diff
@@ -1,106 +1,63 @@
-    let latest_milestone_index = tangle.get_latest_milestone_index();
-    let latest_milestone_timestamp = tangle
+pub(crate) async fn info<B: StorageBackend>(Extension(args): Extension<ApiArgsFullNode<B>>) -> impl IntoResponse {
+    let latest_milestone_index = args.tangle.get_latest_milestone_index();
+    let latest_milestone_timestamp = args
+        .tangle
         .get_milestone(latest_milestone_index)
         .await
         .map(|m| m.timestamp())
         .unwrap_or_default();
 
-    Ok(warp::reply::json(&InfoResponse {
-        name: node_info.name.clone(),
-        version: node_info.version.clone(),
+    Json(InfoResponse {
+        name: args.node_info.name.clone(),
+        version: args.node_info.version.clone(),
         status: StatusResponse {
-            is_healthy: health::is_healthy(&tangle, &peer_manager).await,
+            is_healthy: health::is_healthy(&args.tangle, &args.peer_manager).await,
             latest_milestone_timestamp,
             latest_milestone_index: *latest_milestone_index,
-            confirmed_milestone_index: *tangle.get_confirmed_milestone_index(),
-            pruning_index: *tangle.get_pruning_index(),
+            confirmed_milestone_index: *args.tangle.get_confirmed_milestone_index(),
+            pruning_index: *args.tangle.get_pruning_index(),
         },
         protocol: ProtocolResponse {
-            network_name: network_id.0,
-            bech32_hrp,
-            min_pow_score: protocol_config.minimum_pow_score(),
+            network_name: args.network_name.clone(),
+            bech32_hrp: args.bech32_hrp.clone(),
+            min_pow_score: args.protocol_config.minimum_pow_score(),
             rent_structure: RentStructureResponse {
-                v_byte_cost: protocol_config.byte_cost().v_byte_cost,
-                v_byte_factor_key: protocol_config.byte_cost().v_byte_factor_key,
-                v_byte_factor_data: protocol_config.byte_cost().v_byte_factor_data,
+                v_byte_cost: args.protocol_config.byte_cost().v_byte_cost,
+                v_byte_factor_key: args.protocol_config.byte_cost().v_byte_factor_key,
+                v_byte_factor_data: args.protocol_config.byte_cost().v_byte_factor_data,
             },
         },
         metrics: MetricsResponse {
-            messages_per_second: 0.0,            // TODO
-            referenced_messages_per_second: 0.0, // TODO
-            referenced_rate: 0.0,                // TODO
+            messages_per_second: 0f64,            // TODO
+            referenced_messages_per_second: 0f64, // TODO
+            referenced_rate: 0f64,                // TODO
         },
         features: {
             let mut features = Vec::new();
-            if rest_api_config.feature_proof_of_work() {
+            if args.rest_api_config.feature_proof_of_work() {
                 features.push("PoW".to_string())
             }
             features
         },
         plugins: Vec::new(), // TODO
```

Comment:
Properly document the TODO if it won't make it into this PR (or fix the TODO :trollface: )

Reply:
done 👍